vehicle/mkvi/software/bms/tasks/openwire.c

```diff
@@ -0,0 +1,124 @@
+#include "tasks.h"
+
+#include "vehicle/common/ltc6811/ltc681x.h"
+#include "vehicle/mkvi/software/bms/bms_config.h"
+#include "vehicle/mkvi/software/bms/can_api.h"
+#include "vehicle/mkvi/software/bms/utils/fault.h"
+#include <string.h>
+
+#define NO_OPEN_WIRES (65535)
+#define NUM_CV_REG    (4)
+#define NUM_CHANNELS  (18)
+
+/*
+Reads and parses the LTC681x cell voltage registers.
+The function is used to read the parsed Cell voltages codes of the LTC681x.
+This function will send the requested read commands parse the data
+and store the cell voltages in c_codes variable.
+*/
+uint8_t LTC681x_rdcv_trunc(
+    uint8_t reg, // Controls which cell voltage register is read back.
+    uint8_t total_ic // The number of ICs in the system
+) {
+    int8_t pec_error = 0;
+    uint8_t cell_data_buff[NUM_RX_BYT * total_ic];
+    memset(cell_data_buff, 0x00, NUM_RX_BYT * total_ic);
+
+    if (reg == 0) {
+        for (uint8_t cell_reg = 1; cell_reg < NUM_CV_REG + 1;
+             cell_reg++) // Executes once for each of the LTC681x cell voltage
+                         // registers
+        {
+            LTC681x_rdcv_reg(cell_reg, total_ic, cell_data_buff);
+            for (int current_ic = 0; current_ic < total_ic; current_ic++) {
+                pec_error
+                    = pec_error
+                      + parse_cells(
+                          current_ic, cell_reg, cell_data_buff,
+                          &cell_data.cells[current_ic].cells.c_codes[0],
+                          &cell_data.cells[current_ic].cells.pec_match[0]);
+            }
+        }
+    }
+
+    else {
+        LTC681x_rdcv_reg(reg, total_ic, cell_data_buff);
+
+        for (int current_ic = 0; current_ic < total_ic; current_ic++) {
+            pec_error = pec_error
+                        + parse_cells(
+                            current_ic, reg, &cell_data_buff[8 * current_ic],
+                            &cell_data.cells[current_ic].cells.c_codes[0],
+                            &cell_data.cells[current_ic].cells.pec_match[0]);
+        }
+    }
+    // LTC681x_check_pec(total_ic, CELL, ic);
```

Comment:
we don't actual use the data in this currently, might implement later

Reply:
you already have the pec_errors in a variable. I'd follow the architecture of the other tasks, and return them back to bms.c, where it can handle what to do with them, like this for temps:
```c
    1     // Check for PEC errors
    2     if (rc != 0) {
    3         bms_metrics.temperature_pec_error_count += rc;
    4
    5         if (bms_metrics.temperature_pec_error_count >= MAX_PEC_ERROR_COUNT) {
    6             set_fault(BMS_FAULT_PEC);
    7             bms_core.bms_state = BMS_STATE_FAULT;
    8         }
    9         return;
   10     } else {
   11         bms_metrics.temperature_pec_error_count = 0;
   12     }
```
 you can define `open_wire` in bms.c and pass the pointer into this function, then return `rc` with the number of pecers to be consistent with other tasks 
 
 the description on check_pec says that it updates pec counts in the cell_asic struct - these may be fields we don't have.